spec/index.bs

```diff
@@ -2427,23 +2427,45 @@ used when copying data between a [=texture=] and a [=buffer=].
   - For {{GPUTextureDimension/2d}} textures, data is copied between one [=image=] and one [=array layer=].
   - For {{GPUTextureDimension/3d}} textures, data is copied between one [=image=] and one depth [=slice=].
 
+Issue: Define images more precisely. In particular, define them as being comprised of [=texel blocks=].
+
+Issue: Define the exact copy semantics, by reference to common algorithms shared by the copy methods.
+
 <dl dfn-type=dict-member dfn-for=GPUBufferCopyView>
     : <dfn>bytesPerRow</dfn>
     ::
-        The stride, in bytes, between the beginning of each row of data and
-        the subsequent row.
+        If {{GPUBufferCopyView/bytesPerRow}} is not zero, it is the stride, in bytes, between the
+        beginning of each row of [=texel blocks=] and the subsequent row.
+
+        If {{GPUBufferCopyView/bytesPerRow}} is zero, it is treated as
+        `copySize.width` &divide; [=texel block width=] &times; [=texel block size=] in copy
+        commands.
 
     : <dfn>rowsPerImage</dfn>
     ::
-        {{GPUBufferCopyView/rowsPerImage}} &times; {{GPUBufferCopyView/bytesPerRow}}
+        If {{GPUBufferCopyView/rowsPerImage}} is not zero,
+        {{GPUBufferCopyView/rowsPerImage}} &times; {{GPUBufferCopyView/bytesPerRow}} &divide; [=texel block height=]
         is the stride, in bytes, between the beginning of each [=image=] of data
         and the subsequent [=image=].
 
+        If {{GPUBufferCopyView/rowsPerImage}} is zero, it is treated as `copySize.height` in copy
+        commands.
+
         Note:
         {{GPUBufferCopyView/rowsPerImage}} must be zero for copies with a `copySize.depth` of 1,
         and must be greater than zero otherwise.
 </dl>
 
+<div algorithm class=validusage>
+
+<dfn abstract-op>GPUBufferCopyView Valid Usage</dfn>
+
+Given a {{GPUBufferCopyView}} |bufferCopyView|, the following validation rules apply:
+  - |bufferCopyView|.{{GPUBufferCopyView/buffer}} must be a [=valid=] {{GPUBuffer}}.
+  - |bufferCopyView|.{{GPUBufferCopyView/bytesPerRow}} must be a multiple of 256.
```

Comment:
These rules are the same for both buffer-to-texture and texture-to-buffer copies so I decide to put it here.